subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/LocalComponentDependencyMetadata.java

```diff
@@ -107,7 +107,7 @@ public String getDependencyConfiguration() {
 
     @Override
     public Set<ConfigurationMetadata> selectConfigurations(ComponentResolveMetadata fromComponent, ConfigurationMetadata fromConfiguration, ComponentResolveMetadata targetComponent, AttributesSchema attributesSchema) {
-        assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
+        // assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
```

Comment:
It's not ignoring, AFAIK, because "bytecode length" is one of the first things tested before even looking at the bytecode before inlining. Anyway I agree that for performance tests assertions should be disabled, as they have undesirable side effects.